### PR TITLE
Remove developers table on README.md

### PR DESCRIPTION
**Pull request by rljacob**
Remove the hard-to-maintain developers table and replace
with a alphabetical-by-last-name list of develoeprs in two
catagories: primary and also.

Also change ACME to E3SM

Test suite: none
Test baseline: N/A
Test status: BFB

Fixes ESCMI/cime#1562

Code review:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,48 +1,23 @@
 # cime
 Common Infrastructure for Modeling the Earth
 
-**cime** (pronounced: seem) represents the infrastructure code for the 
+ **cime** (pronounced: seem)providess scripts for configuration, build, and testing of
```
Comment (mvertens) [inline README.md]: I think that the right content here is to put in a more comprehensive statement - like what is contained at the top of the User's guide:
CIME, pronounced “SEAM”, contains the support scripts (configure, build, run, test), data models, essential utility libraries, a “main” and other tools that are needed to build a single-executable coupled Earth System Model. CIME is available in a stand-alone package that can be compiled and tested without active prognostic components but is typically included in the source of a climate model. CIME does not contain: any active components, any intra-component coupling capability (such as atmosphere physics-dynamics coupling).

Comment (rljacob) [reply README.md]: Sure I'll make that change.

**Comment by rljacob** [comment]
It occurred to me that maybe we should list Primary developers by area?
Case Control System, Data Models and MCT-driver.  That covers the big three parts of CIME.  Then the "also contributing" category will include everyone who worked on any of those.

**Comment by mvertens** [comment]
What about the fact that some of the developers have worked in all three
areas (like me?). Would it be odd to have the names repeated?

**Comment by agsalin** [comment]
We need to add back in some attribution of who funded the development. DOE needs to be able to show that it contributed, which is why I added the paragraph and table in the first place. I agree the complexity of the table is no longer needed, and like the simple contributor lists. 

How about adding:
> CIME is jointly developed with support from the Earth System Modeling program of DOE's BER office and <some NCAR/UCAR/NSF attribution>.

File: README.md (hunk 1 above)
Comment (mvertens) [reply README.md]: You did not make the above change . Could you please do that and I can then accept the PR.

**Comment by rljacob** [comment]
@mvertens I think I fixed it.

**Pull request merged**